### PR TITLE
Writing_Documentation.rst: Adds rst tutorial

### PR DESCRIPTION
**Pull request by CosminIcatoiu**
As most of the coala documentation is written in rst
documents, a link to a tutorial on how to actually
do that is added in the document.

Closes https://github.com/coala/coala/issues/3521

### Checklist

- [x] I have rebased properly. Please see [our tutorial on rebasing](http://coala.io/git#rebasing).
- [x] I have gone through the [commit guidelines](http://coala.io/commit) and I've followed them.
- [x] I have followed the [guidelines for the commit shortlog](http://coala.io/commit#shortlog).
- [x] I have followed the [guidelines for the commit body](http://coala.io/commit#commit-body).
- [x] I have included the issue URL with the 'Fixes' keyword if the commit fixes a *real bug* and the 'Closes' keyword if it adds a feature or enhancement. For details, check the [issue reference section in our commit guidelines](http://coala.io/commit#issue-reference).
- [x] I have [run all the tests](http://api.coala.io/en/latest/Developers/Executing_Tests.html) and they all pass. If any CI (below) is not green, please fix them. If GitMate issues appear you will have to amend your commits, pushing new commits on top is not sufficient!

### CODE REVIEW

**Comment by Mixih** [comment]
hmm, `adds`, should be `add` since we want imperative mood.

File: docs/Developers/Writing_Documentation.rst [hunk 1]
```diff
@@ -8,6 +8,11 @@ Documentation is written in reStructuredText and rendered by `Read the Docs
 <https://readthedocs.io>`_ to our lovely users.
 You can view the current user documentation on http://docs.coala.io.
 
+If you have never written a reStructuredText document before or
```
Comment (Mariatta) [inline docs/Developers/Writing_Documentation.rst]: Perhaps more like:

```
coala documentation is written in reStructuredText format. To familiarize yourself with it, please go through this `guide <http://www.sphinx-doc.org/en/latest/rest.html>`_.
```

Comment (CosminIcatoiu) [inline docs/Developers/Writing_Documentation.rst]: The fact that the documentation is written in rst is mentioned before. I think "please go " formulation doesn't sound that friendly. Check the rephrase i did and tell me your opinion. :)

**Comment by CosminIcatoiu** [comment]
How is it ? @Mariatta @Mixih :)

File: docs/Developers/Writing_Documentation.rst [hunk 2]
```diff
@@ -8,6 +8,10 @@ Documentation is written in reStructuredText and rendered by `Read the Docs
 <https://readthedocs.io>`_ to our lovely users.
 You can view the current user documentation on http://docs.coala.io.
 
+To familiarize yourself with the reStructuredText syntax and rules here is
```
Comment (Mixih) [inline docs/Developers/Writing_Documentation.rst]: syntax and rules are more or less equivalent; delete rules

File: docs/Developers/Writing_Documentation.rst [hunk 3]
```diff
@@ -8,6 +8,10 @@ Documentation is written in reStructuredText and rendered by `Read the Docs
...
+a `guide <http://www.sphinx-doc.org/en/latest/rest.html>`_ that may help
```
Comment (Mixih) [inline docs/Developers/Writing_Documentation.rst]: here is -> please see this

File: docs/Developers/Writing_Documentation.rst [hunk 4]
```diff
@@ -8,6 +8,10 @@ Documentation is written in reStructuredText and rendered by `Read the Docs
...
+you.
```
Comment (Mixih) [inline docs/Developers/Writing_Documentation.rst]: delete `that may help you`

**Comment by CosminIcatoiu** [comment]
Done

**Comment by Mariatta** [comment]
ack 5cbd208

**Comment by aptrishu** [comment]
@rultor merge

**Comment by aptrishu** [comment]
ack 5cbd2082d3fe

**Comment by aptrishu** [comment]
@rultor merge

**Pull request merged**